pandas/tests/frame/methods/test_clip.py

```diff
@@ -166,3 +166,13 @@ def test_clip_with_na_args(self, float_frame):
         result = df.clip(lower=t, axis=0)
         expected = DataFrame({"col_0": [9, -3, 0, 6, 5], "col_1": [2, -4, 6, 8, 3]})
         tm.assert_frame_equal(result, expected)
+
+    def test_clip_pos_args_deprecation(self):
+        # https://github.com/pandas-dev/pandas/issues/41485
+        df = DataFrame({"a": [1, 2, 3]})
+        msg = (
+            r"Starting with Pandas version 2\.0 all arguments of clip except "
+            r"for the arguments 'self', 'lower' and 'upper' will be keyword-only"
```

Comment:
I think deprecate_nonkeyword_arguments should be updated first to give a clearer message for methods (i.e. not mention self and maybe also mention the class)

Reply:
yeah, agreed, I'll make a PR to update that first

Reply:
ping on that PR :->